### PR TITLE
bazel: Update googleurl

### PR DESCRIPTION
**Pull request by keith**
This was reverted previously due to an issue with older android NDK versions, the last commit works around that issue until bazel supports newer versions.

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

**Comment by keith** [comment]
@mattklein123 @wrowe @buildbreaker this is the Android only part of the fix (targeting Matt's revert branch) I think with this we should be unblocked to land these

**Review by buildbreaker — DISMISSED** [review]
Yep! This was what I was going for

**Comment by keith** [comment]
Testing on envoy-mobile CI https://github.com/envoyproxy/envoy-mobile/pull/1830

File: bazel/external/googleurl.patch [hunk 1]
```diff
@@ -92,13 +92,29 @@ index 6651220..a469c19 100644
...
  #endif
+diff --git a/base/containers/checked_iterators.h b/base/containers/checked_iterators.h
+index b5fe925..31aa81e 100644
+--- a/base/containers/checked_iterators.h
++++ b/base/containers/checked_iterators.h
+@@ -237,9 +237,11 @@ using CheckedContiguousConstIterator = CheckedContiguousIterator<const T>;
+ // [3] https://wg21.link/pointer.traits.optmem
+ namespace std {
+ 
++#ifdef SUPPORTS_CPP_17_CONTIGUOUS_ITERATOR
```
Comment (mattklein123) [inline bazel/external/googleurl.patch:103]: Can you add some comment here in the patch just to indicate what is going on / what the implication of this is? Also, are we tracking this upstream in any way? cc @RenjieTang @dio @yanavlasov

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Awesome, thanks a ton for tracking this down.

I closed my other PR so can you just bump the dep at the same time and we can review/merge it all at once?

/wait

**Comment by buildbreaker** [comment]
@mattklein123 This PR was intended to merge into your PR 😬

**Review by mattklein123 — APPROVED** [review]
Awesome, thanks!

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -834,13 +834,13 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Chrome URL parsing library",
         project_desc = "Chrome URL parsing library",
         project_url = "https://quiche.googlesource.com/googleurl",
-        # Static snapshot of https://quiche.googlesource.com/quiche/+archive/ef0d23689e240e6c8de4c3a5296b209128c87373.tar.gz.
-        version = "ef0d23689e240e6c8de4c3a5296b209128c87373",
-        sha256 = "d769283fed1319bca68bae8bdd47fbc3a7933999329eee850eff1f1ea61ce176",
+        # Static snapshot of https://quiche.googlesource.com/quiche/+archive/561705e0066ff11e6cb97b8092f1547835beeb92.tar.gz.
```
Comment (dio) [inline bazel/repository_locations.bzl:837]: A small request: 
```suggestion
        # Static snapshot of https://quiche.googlesource.com/googleurl/+archive/561705e0066ff11e6cb97b8092f1547835beeb92.tar.gz.
```

File: bazel/repository_locations.bzl [hunk 3]
```diff
@@ -834,13 +834,13 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
-        version = "ef0d23689e240e6c8de4c3a5296b209128c87373",
-        sha256 = "d769283fed1319bca68bae8bdd47fbc3a7933999329eee850eff1f1ea61ce176",
+        # Static snapshot of https://quiche.googlesource.com/googleurl/+archive/561705e0066ff11e6cb97b8092f1547835beeb92.tar.gz.
+        version = "561705e0066ff11e6cb97b8092f1547835beeb92",
+        sha256 = "7ce00768fea1fa4c7bf658942f13e41c9ba30e9cff931a6cda2f9fd02289f673",
         urls = ["https://storage.googleapis.com/quiche-envoy-integration/googleurl_{version}.tar.gz"],
         use_category = ["controlplane", "dataplane_core"],
         extensions = [],
-        release_date = "2020-07-30",
+        release_date = "2021-03-17",
```
Comment (moderation) [inline bazel/repository_locations.bzl:843]: `2021-08-31` 

https://quiche.googlesource.com/googleurl/+/561705e0066ff11e6cb97b8092f1547835beeb92

Comment (keith) [reply bazel/repository_locations.bzl]: I had this first and it failed and recommended this one (I think), went back to that so we'll see

Comment (moderation) [reply bazel/repository_locations.bzl]: As this is not a Github URL I don't believe we verify the date like we do for the majority of the deps

Comment (keith) [reply bazel/repository_locations.bzl]: Oh the failed is from an unrelated date 🤔

**Comment by moderation** [comment]
Hhhmmm. The error is the incorrect date for `rules_go`

```
github_release:  GitHubRelease(organization='bazelbuild', project='rules_go', version='v0.27.0', tagged=True)
*WARNING* io_bazel_rules_go has a newer release than v0.27.0@<2021-03-18 15:02:42>: v0.28.0@<2021-07-06 23:21:45>
io_bazel_rules_go has a GitHub release date 2021-03-18
An error occurred while processing ./bazel/repository_locations.bzl, please verify the correctness of the metadata: Mismatch with metadata release date of 2021-03-17
```

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Comment by mattklein123** [comment]
Coverage seems like a flake to me.

**Comment by mattklein123** [comment]
Somehow line coverage for lua seems to be persistently failing on this PR which I really don't understand. Can you take a quick look at the line diffs from coverage to see if anything pops up? You can see the coverage report in CI and here is the main report: https://storage.googleapis.com/envoy-postsubmit/main/coverage/index.html. cc @lizan for any ideas on this.

**Comment by keith** [comment]
image

I imagine the tests have some async behavior that causes this to be hit or not, doesn't appear related to this change

**Comment by mattklein123** [comment]
Yup OK I'm fine force merging. We can sort out coverage on main. Thank you!

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**